robottelo/host_helpers/api_factory.py

```diff
@@ -711,6 +712,248 @@ def wait_for_errata_applicability_task(
                 f'No task was found using query " {search_query} " for host id: {host_id}'
             )
 
+    def register_host_and_needed_setup(
+        self,
+        client,
+        organization,
+        activation_key,
+        environment,
+        content_view,
+        enable_repos=False,
+        rex_key=False,
+        force=False,
+        loc=None,
+    ):
+        """
+        Helper will setup desired entities to host content. Then, register the
+        host client to the entities, using associated activation-key.
+
+        Attempt to make needed associations between detached entities.
+
+        Add desired repos to the content-view prior to calling this helper.
+        Or, add them to content-view after calling, then publish/promote.
+
+        The host will be registered to location: None (visible to all locations).
+
+        param client: instance of a rhel contenthost to register.
+        param enable_repos (bool): enable all available repos on the client, after registration?
+            default: False, be sure to enable any repo(s) for client, after calling this method.
+        param rex_key (bool): add a Remote Execution Key to client for satellite?
+            default: False
+        param force (bool): force registration of the client to bypass?
+            default: False, a reused fixture contenthost will fail if already registered.
+        param loc: pass a location object to limit host visibility.
+            default None, client is available to all locations.
+
+        type: Below arguments can be any of the following:
+        - (int) pass id of entity to be read
+        - (str) pass name of entity to be searched
+        - (entity) pass an entity instance
+
+        param organization: pass an Organization instance, name, or id to use.
+        param activation_key: pass an Activation-Key instance, name, or id.
+        param environment: pass a Lifecycle-Environment instance, name, or id.
+            example: can pass string name 'Library'.
+        param content_view: pass a Content-View instance, name, or id.
+            example: can pass string name 'Default Organization View'.
+
+        Notes:
+            1. Will fail if passed entities do not exist in the same organization and satellite.
+            2. Use param `enable_repos`, to try enabling any repositories on client,
+                that were added to content-view prior. But if there are no
+                repositories added/made available, this will fail.
+            3. The 'Default Organization View' cannot be published, promoted, edited, or deleted.
+                but you can register the client to it.
+
+        Steps:
+            1. get needed entities from arguments, id, name, or instance. Read all as instance.
```

Comment:
```suggestion
        Steps:

            1. get needed entities from arguments, id, name, or instance. Read all as instance.
```

Reply:
Locally with this still fails `make-docs` but pushed to see if anything changes for PR checks